valhalla/sif/dynamiccost.h

```diff
@@ -25,6 +25,49 @@
 #include <rapidjson/document.h>
 #include <unordered_map>
 
+// macros aren't great but writing these out for every option is an abomination worse than this macro
+
+/**
+ * this macro takes a ranged_default_t and uses it to make sure user provided values (in json or in
+ * pbf) are in range before setting it on the costing options
+ *
+ * @param costing_options  pointer to protobuf costing options object
+ * @param range            ranged_default_t object which will check any provided values are in range
+ * @param json             rapidjson value object which should contain user provided costing options
+ * @param json_key         the json key to use to pull a user provided value out of the jsonn
+ * @param option_name      the name of the option will be set on the costing options object
+ */
+
+#define JSON_PBF_RANGED_DEFAULT(costing_options, range, json, json_key, option_name)                 \
+  {                                                                                                  \
+    costing_options->set_##option_name(                                                              \
+        range(rapidjson::get<decltype(range.def)>(json, json_key,                                    \
+                                                  costing_options->has_##option_name##_case()        \
+                                                      ? costing_options->option_name()               \
```

Comment:
you will note that the macro is a little bit redundant in that it says if the json isnt there but the pbf is already set, reset it to the value thats already in there, a slight waste of cpu but its trivial imho.